packages/fast-shell-react-msft/src/pane/pane.tsx

```diff
@@ -0,0 +1,324 @@
+        opacity: "0",
+        top: "0",
+        width: "8px",
+        height: "100%",
+        zIndex: "1",
+        transition: "transform .04s ease-in-out",
+        outline: "none",
+        transform: "scale(.5, 1)",
+        "&:hover": {
+            cursor: "ew-resize"
+        },
+        "&:active": {
+            opacity: "1", transform: "scale(1)"
+        }
+    },
+    pane__resizeWest: {
+        "& $pane_resizeHandle": {
+            left: "-4px"
+        }
+    },
+    pane__resizeEast: {
+        "& $pane_resizeHandle": {
+            right: "-4px"
+        }
+    },
+    pane__overlay: {
+        position: "absolute",
+        height: "100%",
+        zIndex: "2"
+    },
+    pane__hidden: {
+        display: "none"
+    }
+};
+
+class Pane extends Foundation<PaneProps, IPaneState> {
+    /**
+     * The default props of the Pane component
+     */
+    public static defaultProps: IPaneHandledProps = {
+        minWidth: 100,
+        maxWidth: 800,
+        resizable: false,
+        collapsed: false,
+        overlay: false,
+        hidden: false,
+    };
+
+    /**
+     * The width of a pane when it is collapsed
+     */
+    private static collapsedWidth: number = 40;
+
+    /**
+     * All handled props
+     */
+    protected handledProps: IPaneHandledProps & IManagedClasses<IPaneClassNamesContract> = {
+        minWidth: void 0,
+        maxWidth: void 0,
+        width: void 0,
+        resizable: void 0,
+        id: void 0,
+        collapsed: void 0,
+        overlay: void 0,
+        hidden: void 0,
+        resizeFrom: void 0,
+        managedClasses: void 0
+    };
+    /**
+     * Stores a reference to the pane HTML element
+     */
+    private rootElement: React.RefObject<HTMLDivElement>;
+
+    constructor(props: PaneProps) {
+        super(props);
+
+        this.state = {
+            resizing: false,
+            dragReference: null,
+            width: 300
+        };
+
+        this.onMouseMove    = throttle(this.onMouseMove, 16);
+        this.onWindowResize = rafThrottle(this.onWindowResize);
+        this.rootElement = React.createRef();
+    }
+
+    /**
+     * Return the width of Pane. Sources from props first, and then state if props.width is undefined
+     */
+    public width(): number {
+        return this.props.width || this.state.width;
+    }
+
+    /**
+     * Handle when component is mounted to the DOM
+     */
+    public componentDidMount(): void {
+        window.addEventListener("resize", this.onWindowResize);
+    }
+
+    /**
+     * Handle when component is removed from the DOM
+     */
+    public componentWillUnmount(): void {
+        window.removeEventListener("resize", this.onWindowResize);
+    }
+
+    /**
+     * Handle when component updates
+     */
+    public componentDidUpdate(prevProps: PaneProps, prevState: IPaneState): void {
+        if ( this.state.resizing && !prevState.resizing ) {
+            document.addEventListener("mouseup", this.onMouseUp);
```

Comment:
Do we need to check for the DOM on these prior to accessing the document? Assume we would need a check when adding event listeners above as well?

suggest https://www.npmjs.com/package/exenv-es6 :)